pages/blog/intent-driven-api.md

```diff
@@ -0,0 +1,254 @@
+
+## Designing APIs that are resilient to breaking changes
+
+APIs are meant to solve user needs. 
+
+However, we do not listen to the final users that much. Instead, we tend to expose functionalities based on our own experience, biased by our position or the knowledge we own of the platform behind. Sometimes guided by a vague research (Product-oriented) or lead with a not so clear goal in mind. At least I used to do it in that way.
+
+### Be ready for what the user needs
+
+Focusing on what the user really needs pulls away most of the overprint of any API: What is the user **intention** when asking for a particular action to happen?
+
+An Intent represents a user intention of performing an action that solves a clear use case.
+
+Continuing with the metaphor presented in the previous paragraph, my only intention by then was to **buy groceries** **for the dinner**. In particular, I needed some avocados, tomatoes and a baguette.
+
+The supermarket should provide a mechanism that let me buy those items. How to get those, it's just the implementation detail and can be up to each supermarket to decide how to do it. 
+
+For example, my supermarket had shelfs where fruits can be picked up from, offers bags, a balance to know the weight of those and a checkout place. However, after the last changes, fruits are weighted at checkout and baguettes are no longer in the shelf. You need to ask the baker. 
+
+As you can see, the interface didn't change: I still can buy the groceries. However, the implementation changed completely.
+
+Building APIs based on the implementation detail will nothing less than lead your users to suffer the (bad) design choices made in the past. Each change you do, will penalize completely the user experience and force to upgrade their code.
+
+Let's pick up another example, this time a more practical one. 
+
+Let's use our [AsyncAPI Parser](https://github.com/asyncapi/parser-js). Built in Javascript, it parsers an AsyncAPI spec doc and provides a set of functions to work with the document and access the different objects and their values.
+
+Here is a mod version of the [Streetlights tutorial](https://www.asyncapi.com/docs/tutorials/streetlights) document made for demo purposes:
+
+```yaml
+asyncapi: '2.0.0'
+info:
+  title: Streetlights mod API
+  version: '1.0.0-alpha'
+servers:
+  mosquitto:
+    url: mqtt://test.mosquitto.org
+    protocol: mqtt
+channels:
+  light/measured:
+    publish:
+      summary: Inform about environmental lighting conditions for a particular streetlight.
+      operationId: onLightMeasured
+      message: 
+        $ref: "#/components/messages/lightMessage"
+      
+  light/measured/changed:
+    subscribe:
+      summary: Receive an update every time a lighting condition changed.
+      operationId: onLightMeasureChanged
+      message: 
+        $ref: "#/components/messages/lightMessage"
```

Comment:
```suggestion
  light/measured/changed:
    subscribe:
      summary: Receive an update every time a lighting condition changed.
      operationId: onLightMeasureChanged
      message: 
        $ref: "#/components/messages/lightMessage"
...
```
Since this last channel does not provide anything to the example maybe we should hide it? 🤔

Reply:
Fixed!